Compilation error in `transmute-optimized.rs`

https://github.com/CHERIoT-Platform/cheri-rust/blob/9ffa5750c3d69192e8c2df73d72e70afb0b45020/tests/codegen-llvm/transmute-optimized.rs#L35-L49

Fails, when compiling to CHERIoT, with:
```
error[E0512]: cannot transmute between types of different sizes, or dependently-sized types
  --> local/tests/codegen-llvm/transmute-optimized.rs:40:34
   |
40 |     let p: *const i32 = unsafe { core::mem::transmute(x) };
   |                                  ^^^^^^^^^^^^^^^^^^^^
   |
   = note: source type: `NonZero<usize>` (32 bits)
   = note: target type: `*const i32` (64 bits)

error[E0512]: cannot transmute between types of different sizes, or dependently-sized types
  --> ocal/tests/codegen-llvm/transmute-optimized.rs:48:29
   |
48 |     let a: isize = unsafe { core::mem::transmute(x) };
   |                             ^^^^^^^^^^^^^^^^^^^^
   |
   = note: source type: `NonNull<i32>` (64 bits)
   = note: target type: `isize` (32 bits)

error: aborting due to 2 previous errors
```

--- a/tests/codegen-llvm/transmute-optimized.rs
+++ b/tests/codegen-llvm/transmute-optimized.rs
@@ -39,7 +39,7 @@ pub fn non_null_is_null(x: core::ptr::NonNull<i32>) -> bool {
 #[no_mangle]
 pub fn non_zero_is_null(x: core::num::NonZero<usize>) -> bool {
     // CHECK: ret i1 false
-    let p: *const i32 = unsafe { core::mem::transmute(x) };
+    let p: *const i32 = core::ptr::without_provenance(x.get());
     p.is_null()
 }
 
@@ -47,7 +47,7 @@ pub fn non_zero_is_null(x: core::num::NonZero<usize>) -> bool {
 #[no_mangle]
 pub fn non_null_is_zero(x: core::ptr::NonNull<i32>) -> bool {
     // CHECK: ret i1 false
-    let a: isize = unsafe { core::mem::transmute(x) };
+    let a: usize = x.as_ptr().addr();
     a == 0
 }
 

